pkg/aks/create.go

```diff
@@ -148,17 +148,23 @@ func CreateOrUpdateCluster(ctx context.Context, cred *Credentials, clusterClient
 		}
 	}
 
-	managedCluster := containerservice.ManagedCluster{
-		Name:     to.StringPtr(spec.ClusterName),
-		Location: to.StringPtr(spec.ResourceLocation),
-		Tags:     tags,
-		ManagedClusterProperties: &containerservice.ManagedClusterProperties{
-			KubernetesVersion: spec.KubernetesVersion,
-			AgentPoolProfiles: &agentPoolProfiles,
-			LinuxProfile:      linuxProfile,
-			NetworkProfile:    networkProfile,
-			AddonProfiles:     addonProfiles,
-		},
+	// Check if a managedCluster already exists in Azure. If so, then pull the configuration for that cluster instead
+	// of creating a brand new one.
+	managedCluster, err := clusterClient.Get(ctx, spec.ResourceGroup, spec.ClusterName)
+
+	if err != nil {
```

Comment:
Discussed this offline. 2 items to address:
- Improve error handling here, we cannot do the same here regardless of kind of error we get here.
- With setting of certain fields like `ManagedClusterProperties.KubernetesVersion` moved to under `if` here, these will no longer be updated in case we didn't get an error on `clusterClient.Get` and that would result in a regression where changes done to these fields in Rancher would not propagate down to Azure.

Reply:
* Improved error handling
* Right-this data had to be merged together so as to allow changes in Rancher to propagate to Azure without overwriting other pre-existing values in Azure. I refactored most of `create.go` to separate out the Create and Update functionality (still in progress but what is mentioned above here is fixed. I can add specifically adding `KubernetesVersion` to my testing)

Reply:
Apologies for the earlier force push. I've added a new commit for my changes today